src/main/java/org/cactoos/set/SetOf.java

```diff
@@ -60,7 +61,7 @@ public SetOf(final T... array) {
     public SetOf(final Iterable<T> src) {
         super(() -> {
             final Set<T> tmp = new HashSet<>();
-            new And(tmp::add, src)
+            new And((Proc<T>) tmp::add, src)
```

Comment:
@fanifieiev I think it would be way better to use `src.forEach(tmp::add)`, that's what `Iterable` is made for.

Reply:
@victornoel Sure, that one looks better. But just wondering what was the `And(final Proc<X> proc, final Iterable<X> src)` created for ?

Reply:
@fanifieiev I think it is here only for completeness. `And` is more about the logical assertion (which does not rely on side effect as the `Proc` version does).

I created #1226 in reaction to this discussion :)